README.md

```diff
@@ -0,0 +1,41 @@
+# Zync - the sync tool
+
+Zync is going to take your 3scale data and push them somewhere else, reliably.
+
+## Terminology
+
+Before describing the design it is important to define the terms.
+
+* **Zync** - deployment of this project, Web API.
+
+
+* **3scale** - 3scale API Manager (system).
+* **Tenant** - mapping of Provider id registered in 3scale API Manager to domain and access token.
+* **Model** - relevant object in **3scale** like Application, Limit, Metric, Service.
+* **Notification** - Message sent to **Zync** descibing the **model** that changed and having all required properties to fetch it from the API later.
+  * Example: Application 3456, tenant_id: 26
+  * Example: Limit 4566, metric_id: 36, application_id: 46, tenant_id: 16
+* **Update** - **Zync** fetches updated **Model** from the **Notifier** via the API using the information provided by the **Notification**.
+* **Lock** - mechanism that prevents two pieces of code to be executing in parallel for the same scope
+  * Example: **Tenant Lock** would mean only one can be running for one **Tenant**
+  * Example: **Model** **Lock** - only one per uniquely identified **Model**
+* **Entry** - The information from the API provided by the **Update**.
+* **Log** - ordered list of **Entries** as they were received
+* **Integration** - code that pushes **Log** **entries** one-by-one for the same **Model** to some external service
+
+## Design
+
+**Zync** is meant to synchronize data from **3scale** to external systems (like IDPs). Some people use for this Webhooks but without further logic they can be unreliable and arrive out of order. This tool is meant to synchronize the final state into different system.
```

Comment:
grammatical suggestions :) - `Some people use Webhooks for this` ... `state to a different system`